source/exe/win32/service_base.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <functional>
+#include <string>
+
+#include "envoy/event/signal.h"
+
+#include "exe/service_status.h"
+
+namespace Envoy {
+class ServiceBase {
+public:
+  virtual ~ServiceBase() = default;
+
+  static bool TryRunAsService(ServiceBase& service);
```

Comment:
Could probably use some comments on what the various methods are for, e.g. TryRunAsService registers ServiceMain etc.

Reply:
See the updated comments in most functions.